R/FilteredData.R

```diff
@@ -875,7 +881,19 @@ FilteredData <- R6::R6Class( # nolint
 
             header_html <- tags$tr(
               tagList(
-                lapply(colnames(datasets_df), tags$td)
+                lapply(
+                  vapply(
+                    seq_along(datasets_df),
```

Comment:
In a for loop we could also add a warning, if `label` is missing.

Reply:
re: integers
I replied in the other comment. I need it to do `names(df)[[i]]`.

re: for vs vapply
In this case, I have found `vapply` more handy because on each iteration I am appending to the result vector.
```
res <- character(0)
for (i in x) {
  res_i <- foo(i)
  res <- append(res, res_i)
}
```
vs
```
vapply(
  x
  \(i) foo(i)
)
```


re: warning
I have been thinking about it but I decided to stick to the current behaviour and focus on fixing the error and not to enhance the logic. I didn't want to make this PR big (even though I did it few times but this is nothing good IMHO)

Reply:
In `for` loop you can bind/concatenate/append with `c` function.
And I meant one `for` instead of a joint usage of `lapply(vapply)`. 

Instead of

```r
lapply(vapply(x, fun1), fun2)
```

do

```r
for(i in x)
x_res <- c(x_res, fun2(fun1(i))
```

or maybe you could do 

```r
vapply(x, function(i) fun2(fun1(i))
```

What do you think?

Reply:
Regardless of the function used - `vapply` gives this to you out-of-box.  If I don't need to capture the result for each iteration (e.g. loop to trigger side-effects) then I would probably choose `for` loop instead. But here I do, that's why I think it's handy.

`lapply(vapply(...))` was introduced to increase the readability - I separated label derivation out of html. Two loops instead of one but not nested - still of linear complexity (2O(n)).

Both are good and fully functional. It's just a balance between readability vs complexity. I just wanted this to be easily readable. Maybe I got to far? I don't know :)

Reply:
Ok, I think I'm convinced with the final output. `vapply` over `seq_along` gives you the benefit of concatenated output that would be a vector (not a list in comparison to lapply). In the end I see we do not use `lapply` for `html` derivation so there's no gonna be any `lapply(vapply`. Also the `ifelse` statement is simplfied. Thanks for the extra work. Closing